heat/sparse/dcsr_matrix.py

```diff
@@ -337,3 +337,22 @@ def __repr__(self) -> str:
         if self.comm.rank != 0:
             return ""
         return print_string
+
+    def __str__(self) -> str:
```

Comment:
It would be better if this function is moved to the `core/printing` module. There are cases like local print vs global print to be handled. See the `__str__` method of the `DNDarray` class for more information.

And it is usually best to use the printing format used by PyTorch since it gives more information to the user. The information outputted by `__str__` should prioritize user-readability more than anything else. Just the `indptr` and `indices` would be too difficult to understand. I, personally, would prefer to see the exact coordinates being printed out. I say this because the users of Heat shouldn't be expected to know about the CSR matrix and its format. They should just be able to use this more-efficient data structure without having to learn too much about it.

Reply:
What @mrfh92 says also seems acceptable to me. Maybe the users won't want to see the actual data points. But my only problem with this solution is, the users wouldn't be able to see the data points even if they wanted to. There is no other way for them to get the coordinates if we don't show it in the `print` output.